test/switchboard/default-switchboards/FastSwitchboard.t.sol

```diff
@@ -5,104 +5,305 @@ import "../../Setup.t.sol";
+            root,
+            packetId,
+            1,
+            _a.chainSlug,
+            0
+        );
+
+        assertTrue(isAllowed);
     }
 
-    function testIsAllowed() external {
+    function testRecoveryFromWrongProposal() external {
+        bytes32 invalidRoot = bytes32(uint256(100));
+        signAndPropose(_b.chainSlug, packetId, invalidRoot); // wrong root
+        uint256 proposalCount = 1; // this is second proposal, 1st one proposed in setup
+
         bytes32 digest = keccak256(
-            abi.encode(address(fastSwitchboard), _a.chainSlug, packetId)
+            abi.encode(
+                TRIP_PROPOSAL_SIG_IDENTIFIER,
+                address(fastSwitchboard),
+                _a.chainSlug,
+                _b.chainSlug,
+                fastSwitchboard.nextNonce(_watcher),
+                packetId,
+                proposalCount
+            )
         );
         bytes memory sig = _createSignature(digest, _watcherPrivateKey);
 
-        fastSwitchboard.attest(packetId, sig);
+        fastSwitchboard.tripProposal(
+            fastSwitchboard.nextNonce(_watcher),
+            packetId,
+            proposalCount,
+            sig
+        );
 
-        digest = keccak256(
-            abi.encode(address(fastSwitchboard), _a.chainSlug, packetId)
+        assertTrue(fastSwitchboard.isProposalTripped(packetId, proposalCount));
+
+        signAndPropose(_b.chainSlug, packetId, root); // wrong root
+        proposalCount = 2;
+
+        _attestOnDst(
+            address(fastSwitchboard),
+            _b.chainSlug,
+            packetId,
+            proposalCount,
+            _watcherPrivateKey
+        );
+
+        _attestOnDst(
+            address(fastSwitchboard),
+            _b.chainSlug,
+            packetId,
+            proposalCount,
+            _altWatcherPrivateKey
+        );
+
+        bool isAllowed = fastSwitchboard.allowPacket(
+            root,
+            packetId,
+            proposalCount,
+            _a.chainSlug,
+            0
         );
-        sig = _createSignature(digest, _altWatcherPrivateKey);
 
-        fastSwitchboard.attest(packetId, sig);
+        assertTrue(isAllowed);
+    }
+
+    function testIsAllowed() external {
+        _attestOnDst(
+            address(fastSwitchboard),
+            _b.chainSlug,
+            packetId,
+            0,
+            _watcherPrivateKey
+        );
+        _attestOnDst(
+            address(fastSwitchboard),
+            _b.chainSlug,
+            packetId,
+            0,
+            _altWatcherPrivateKey
+        );
 
         uint256 proposeTime = block.timestamp -
             fastSwitchboard.timeoutInSeconds();
 
         bool isAllowed = fastSwitchboard.allowPacket(
+            root,
+            packetId,
+            0,
+            _a.chainSlug,
+            proposeTime
+        );
+
+        assertTrue(isAllowed);
+    }
+
+    function testAttestationOnMultipleProposal() external {
```

Comment:
Need test cases where 2 watchers attest different proposalCount.
In case they are same root, allowPacket should be true for both.
In case they are different packats and time has not passed, allowPacket should be false for both.